File: api/src/main/java/org/pmiops/workbench/compliance/ComplianceService.java
package org.pmiops.workbench.compliance;

import java.util.Map;
import org.pmiops.workbench.moodle.ApiException;
import org.pmiops.workbench.moodle.model.BadgeDetailsV2;

public interface ComplianceService {

  enum BadgeName {
    REGISTERED_TIER_TRAINING,
    CONTROLLED_TIER_TRAINING
  }

  /**
   * Get details about the Research Ethics Training and the Data Use Agreement badges for a user
   *
   * @param email
   * @return map of badge name to badge details
   * @throws ApiException
   */
  Map<BadgeName, BadgeDetailsV2> getUserBadgesByBadgeName(String email) throws ApiException;
}

File: api/src/main/java/org/pmiops/workbench/compliance/ComplianceService.kt
package org.pmiops.workbench.compliance

import org.pmiops.workbench.moodle.ApiException
import org.pmiops.workbench.moodle.model.BadgeDetails

interface ComplianceService {

    /**
     * Get Moodle Id associated with Aou User email id
     *
     * @param email
     * @return Moodle Id
     * @throws ApiException
     */
    @Throws(ApiException::class)
    fun getMoodleId(email: String): Int?

    /**
     * Get the list of badges earned by User
     *
     * @param userMoodleId
     * @return list of badges/completed training by user
     * @throws ApiException
     */
    @Throws(ApiException::class)
    fun getUserBadge(userMoodleId: Int): List<BadgeDetails>
}
